### PR TITLE
Deprecate self-references in CREATE clauses

### PR DESCRIPTION
**Pull request by hedengran**

### CODE REVIEW

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 1]
```diff
@@ -16,6 +16,31 @@ New features are added to the language continuously, and occasionally, some feat
 This section lists all of the features that have been removed, deprecated, added, or extended in different Cypher versions.
 Replacement syntax for deprecated and removed features are also indicated.
 
+[[cypher-deprecations-additions-removals-5.8]]
```
Comment (sherfert) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
[[cypher-deprecations-additions-removals-5.9]]
```

**Comment by neo-technology-commit-status-publisher** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-docs-cypher-607.surge.sh

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 2]
```diff
@@ -16,6 +16,31 @@ New features are added to the language continuously, and occasionally, some feat
...
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CREATE (a {foo:1}), (b {foo:a.foo})
+----
+a|
+
+Creating a node or relationship entity,
+and then referencing a property of that entity in the same create clause is deprecated.
```
Comment (sherfert) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
and then referencing that entity in the same create clause is deprecated.
```

This is not really about the property part, right? Any other reference, e.g. `labels(a)` would also be deprecated?

**Review by sherfert — COMMENTED** [review]
Sorry, I had this pending for a while but forgot to submit the review.

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc (hunk 2 above)
Comment (hedengran) [reply modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]: You're right, "referencing a property" does not catch it all, forgot about those other cases. But some references are still valid, e.g. `CREATE (a)-[:REL]->(a)`. Not sure how to capture that in one sentence.

Comment (sherfert) [reply modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]: Does "referencing that entity in a property definition in the same create clause ..." catch all the cases?

Comment (hedengran) [reply modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]: Yeah I think that works, thanks!

**Review by sherfert — APPROVED** [review]

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 3]
```diff
@@ -19,6 +19,29 @@ Replacement syntax for deprecated and removed features are also indicated.
...
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CREATE (a {foo:1}), (b {foo:a.foo})
+----
+a|
+
+Creating a node or relationship entity,
+and then referencing that entity in a property definition in the same create clause is deprecated.
+Please split your `CREATE` into two separate clauses instead.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
and then referencing that entity in a property definition in the same `CREATE` clause is deprecated.
Split the `CREATE` clause into two separate clauses instead.
```

**Review by JPryce-Aklundh — CHANGES_REQUESTED** [review]
looks good overall 👍 - I made two suggestions:

**Review by JPryce-Aklundh — APPROVED** [review]

**Pull request merged**